.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,52 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '35 12 * * 1'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
```

Comment:
I don't think these are strictly required for public repos, like `circulation` and `circulation-core`, but I don't think there's any harm in having them.